docs/DeveloperGuide.md

```diff
@@ -1,34 +1,565 @@
+
+`XYZParser` will then parse the parameter and perform input validation, before creating the Command object to return to `Main`.
+
+<br/>
+
+### 3.4 Command Component
+![Command Class Diagram](images/Command-Class-Diagram.png) <br/>
+*Figure 5: Command Class Diagram*
+
+The `Command` component consists of different commands represented together as `XYZCommand` which all inherits from the abstract `Command` class. 
+
+Example: The Browse command would be represented by a `BrowseCommand`.
+
+`Main` would utilise the `Command.execute` operation to carry out the execution of the command and retrieve a String output that will contain the successful result of the `Command`. If the `Command` was not successful an exception will be thrown with details of the failure.
+
+<br/>
+
+### 3.5 AnimeData Component
+![AnimeData Class Diagram](images/AnimeData-Class-Diagram.png) <br/>
+*Figure 6: AnimeData Class Diagram*
+
+The `AnimeData` component is responsible for retrieving offline json data and parsing it into `Anime` objects that will be stored in program memory. The `AnimeData` will manage an ArrayList of `Anime` objects providing AniChan with an interface for the program to retrieve with the source data.
+
+The `AnimeData `component:
+*   can retrieve Anime objects using ID.
+*   can view detailed  information of each Anime Object.
+*   can browse the Anime catalog with sorting algorithms.
+
+<br/>
+
+<!-- @@author ChanJianHao -->
+### 3.6 User Component
+![User Class Diagram](images/User-Class-Diagram.png) <br/>
+*Figure 7: User Class Diagram*
+
+The User inherits from the abstract `Human` class and stores the name and gender of the user. It represents the user's interaction with `Workspace` class.
+
+The `User`component: 
+*   can provide user information like `name`, `gender`, and `honorific name`
+*   stores an array list of type `Workspace`
+*   can add, set, and switch between workspaces 
+
+The `Workspace` component:  
+*   can allow `User` to create and get the list of `Watchlist` and `Bookmark`.
+*   can allow `User` to change his active `Watchlist`.
+
+<!-- @@author -->
+<br/>
+
+### 3.7 Storage Component
+![Storage Class Diagram](images/Storage-Class-Diagram.png) <br/>
+*Figure 8: Storage Class Diagram*
+
+The `Storage` component consist of `StorageManager` which:
+*   can **save** workspace created by the user as a folder.
+*   can **save** user, watchlist and bookmark data in `.txt` format and **read it back** using their respective storage class, `UserStorage`, `WatchlistStorage`, and `BookmarkStorage`.
+*   can **read** script files that are in `.txt` format using the class `ScriptStorage`.
+
+**AniChan** saved these data as `.txt` files so advanced users will be able to view and manipulate these saved data easily with any available text editor.
+
+<br/>
+
+## 4. Implementation
+This section introduces the specific implementation details and design consideration of some features in AniChan.
+
+<br/>
+
+### 4.1 Estimation Feature
+
```

Comment:
![2](https://user-images.githubusercontent.com/60357492/97390305-6d158080-1917-11eb-86f7-6ac666565ec5.PNG)

This portion seems overly confusing to a new user like me... i think a flow diagram/class diagram would greatly help with explaining this implementation.